### PR TITLE
Migrate maven_jar usages to rules_jvm_external

### PR DESCRIPTION
**Pull request by jin**
`maven_jar` is planned for removal in Bazel 2.0. 

This pull request converts all maven_jar usages to https://github.com/bazelbuild/rules_jvm_external.

See https://github.com/bazelbuild/bazel/issues/6799 for more information about `maven_jar`'s removal, and [this document](https://docs.google.com/document/d/1CqxPv-TwvFWBHgtg7_QhVfck4vH4rnmCU_IuMdC7O2M/edit#) for the design behind the migration tool used for this PR.

### CODE REVIEW

**Comment by shahms** [comment]
It appears as though you're not defining the `rule_jvm_external` repository prior to trying to load it?

**Comment by jin** [comment]
Tests are passing, PTAL.

File: WORKSPACE [hunk 1]
```diff
@@ -65,5 +65,49 @@ ts_setup_workspace()
...
+
+RULES_JVM_EXTERNAL_TAG = "2.9"
+RULES_JVM_EXTERNAL_SHA = "e5b97a31a3e8feed91636f42e19b11c49487b85e5de2f387c999ea14d77c7f45"
+
+http_archive(
+    name = "rules_jvm_external",
+    strip_prefix = "rules_jvm_external-%s" % RULES_JVM_EXTERNAL_TAG,
+    sha256 = RULES_JVM_EXTERNAL_SHA,
+    url = "https://github.com/bazelbuild/rules_jvm_external/archive/%s.zip" % RULES_JVM_EXTERNAL_TAG,
+)
```
Comment (shahms) [inline WORKSPACE]: This part should go in `setup.bzl`

File: WORKSPACE [hunk 2]
```diff
@@ -65,5 +65,49 @@ ts_setup_workspace()
...
+    ],
+    repositories = [
+        "https://jcenter.bintray.com",
+        "https://maven.google.com",
+        "https://repo1.maven.org/maven2",
+    ],
+    fetch_sources = True,
+    generate_compat_repositories = True, # Required by bazel-common's dependencies
+    version_conflict_policy = "pinned",
+)
```
Comment (shahms) [inline WORKSPACE]: This part should replace the `maybe(native.maven_jar, ...)` in `external.bzl`

File: WORKSPACE [hunk 3]
```diff
@@ -65,5 +65,49 @@ ts_setup_workspace()
...
+load("@maven//:compat.bzl", "compat_repositories")
+compat_repositories()
```
Comment (shahms) [inline WORKSPACE]: And these two lines should stay here.

File: WORKSPACE (hunk 1 above)
Comment (jin) [reply WORKSPACE]: Done.

File: WORKSPACE (hunk 2 above)
Comment (jin) [reply WORKSPACE]: Done.

**Pull request merged**